### DISCUSSION TITLE
Protótipo para reduzir a complexidade do setup de infraestrutura para novos projetos

### DISCUSSION THREAD

**Discussion by jpaulo2828** [frontendbr/forum#2054] [Mostre seu trabalho]
Bom dia a todos! 

Em entrevistas com alguns desenvolvedores full-stack, vi que o setup de infraestrutura otimizada para novos projetos acaba sendo uma tarefa árdua que consome bastante tempo. 

Como sempre existe aquela pressão por entrega de features, também não sobra muito tempo para automatização do ambiente de desenvolvimento com ferramentas como CI/CD, formatter, linter e framework’s de teste. 

Bom, comecei a pensar se não seria possível criar um **self-service de tecnologias** em que se escolhe sua stack em alguns click’s e já faz o download do codebase pronto para produção. 

**Como pensei esse fluxo de montar seu codebase:**
🧤 Linguagem para back/front & frameworks
☁️ Cloud provider
💻 Plug cloud services adicionais (ex. base de dados)

...e faz o download instantâneo do seu bootstrapped codebase 📦

Acreditam que faz sentido? Sentem essa dor? Comecei a rabiscar um protótipo e já estou pensando em como executar… Seria muito rico ter o feedback de vocês.

**Protótipo no link:**
https://buffet.stackforge.tech/?referrer={fontend_git}

Grande abs e boa semana a todos!

**Comment by joaroque** [comment]
Obrigado

**Comment by wilsonneto-swe** [comment]
Os CRUDs em qualquer linguagem são bem simples, morosos mas simples, não chegaria a chamar de dor. Mas em projetos com o foco nas regras de negócios em vez de apenas CRUDs acho muito díficil algo assim, até pq cada cenário teria sua estrutura conforme as demandas, necessidades e regras de negócios. Enfim, penso que para CRUDs seria legal, mas para softwares pensados nos usecases e no domínio, poderia ser um pouco complicado... e eu prefiro no segundo cenário fazer tudo desde a base puxando das demanas e necessidades do negócio. (Azure, AWS, DO, functions, serverless, headless, db sql, db nosql, k8s, contaner, apenas código, redis, memcache, cassandra, Cosmos, Storage, combinações entre 2,3,4 destes?...)

**Comment by ninetails** [comment]
Setup de ambiente é algo pessoal, que varia de dev pra dev ou até mesmo de empresa pra empresa dependendo dos standards combinados pela equipe.

Acho que a melhor alternativa é realmente ter um tooling bem feito, mas nenhuma empresa pequena, média ou grande convencionalmente faz isso. Das empresas que eu tenho conhecimento, sei somente da SumUp quando fizeram o Design System deles, se preocuparam pra tooling em frontend.

Mas pq eu digo tooling bem feito? Não digo tooling só ter linter, test e CI/CD configurados, e sim uma forma rápida e eficiente, componentizada, extensível e escalável de auxiliar no desenvolvimento. Quando falo nisso, eu estou dizendo: templates e codemods.

Uma ferramenta bem bacana que cheguei a pesquisar sobre é o [mrm](https://mrm.js.org/). E ferramentas de templates, não há melhor quando se fala de Angular, com o Angular CLI... Quer criar um controller? Só mandar na linha de comando e tal.

Mas mesmo assim, não encontrei nenhuma solução universal.

---

Eu não encontrava esse **mrm** em nenhum lugar auehueahu lembrava que existia. Consegui ver o nome dela a partir deste post do SurvievJS: https://survivejs.com/maintenance/infrastructure/automation/
